pkg/cmd/stack/deploy.go

```diff
@@ -66,12 +66,18 @@ type analyticsTrackerInterface interface {
 	TrackImageBuild(meta ...*analytics.ImageBuildMetadata)
 }
 
+// Divert is the interface for the divert operations needed for stacks command
+type Divert interface {
+	UpdatePod(spec apiv1.PodSpec) apiv1.PodSpec
+}
+
 // Stack is the executor of stack commands
 type Stack struct {
 	K8sClient        kubernetes.Interface
 	Config           *rest.Config
 	AnalyticsTracker analyticsTrackerInterface
 	IoCtrl           *io.Controller
+	Divert           Divert
```

Comment:
Added this to apply the divert translation as part of the compose logic as compose aren't deployed through the proxy anymore